app/server/lib/DocApi.ts

```diff
@@ -1222,7 +1225,7 @@ export class DocWorkerApi {
 
     this._app.get('/api/docs/:docId/download/table-schema', canView, withDoc(async (activeDoc, req, res) => {
       const doc = await this._dbManager.getDoc(req);
-      const options = this._getDownloadOptions(req, doc.name);
+      const options = await this._getDownloadOptions(req);
```

Comment:
This now results in `getDoc` being called twice. Maybe we can thread the `doc` down to _getDownloadFilename, which can optionally use it in place of making a fresh query?